### ISSUE TITLE
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2017-13-index  

### ISSUE THREAD

**Issue by muxueo** [xnl-h4ck3r/waymore#25]
Hi, I use waymore
command line is 
```
python3 waymore.py -i target.com -mode U -v 
```

show 504

![image](https://github.com/xnl-h4ck3r/waymore/assets/50110633/ef657bac-befd-4135-8ef1-ba175fd252f9)

```
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2017-13-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2016-07-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2017-39-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2020-34-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2013-48-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2018-43-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2020-50-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2018-39-index
[ 504 ] Error for https://index.commoncrawl.org/CC-MAIN-2020-05-index   
```

**Comment by xnl-h4ck3r** [comment]
Hi @muxueo 
There seem to be a lot of issues with the Common Crawl API recently. I did email them a while back and they said the following:
```
We're sorry about this. Currently, there is no way to reallyprevent timeouts generally:
- right now, the URL index server is able to handle about 100k look ups per hour.
- however, the number of requests regularly exceeds 1 Mio. per hour  which inevitably causes timeouts
- the number of served requests per IP is limited, but apparently  there are users which have a large IP range at hand or even
 a proxy network at hand
- in short, it's difficult to still allow for bulk requests without allowing single users to consume most of the bandwidth

We plan to move the server to a more capable hardware during summer.
But also then it will definitely not allow for an unlimited number of requests.
```

For now, you can exclude the Common Crawl source altogether using tyhe `-xcc` flag.
Or, you could just check a few of the most recent "indexes" instead of all 95. For example, using `-lcc 3` check the 3 most recent indexes. 

Regards
Xnl

**Comment by muxueo** [comment]
Thank you. I will close this issue now
